src/main/java/cloud/orbit/spring/actuate/ActorInfoContributorLifetimeExtension.java

```diff
@@ -0,0 +1,176 @@
+/*
+ Copyright (C) 2017 Electronic Arts Inc.  All rights reserved.
+
+ Redistribution and use in source and binary forms, with or without
+ modification, are permitted provided that the following conditions
+ are met:
+
+ 1.  Redistributions of source code must retain the above copyright
+     notice, this list of conditions and the following disclaimer.
+ 2.  Redistributions in binary form must reproduce the above copyright
+     notice, this list of conditions and the following disclaimer in the
+     documentation and/or other materials provided with the distribution.
+ 3.  Neither the name of Electronic Arts, Inc. ("EA") nor the names of
+     its contributors may be used to endorse or promote products derived
+     from this software without specific prior written permission.
+
+ THIS SOFTWARE IS PROVIDED BY ELECTRONIC ARTS AND ITS CONTRIBUTORS "AS IS" AND ANY
+ EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ DISCLAIMED. IN NO EVENT SHALL ELECTRONIC ARTS OR ITS CONTRIBUTORS BE LIABLE FOR ANY
+ DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package cloud.orbit.spring.actuate;
+
+import org.springframework.boot.actuate.info.Info;
+import org.springframework.boot.actuate.info.InfoContributor;
+
+import cloud.orbit.actors.Actor;
+import cloud.orbit.actors.extensions.LifetimeExtension;
+import cloud.orbit.actors.runtime.AbstractActor;
+import cloud.orbit.concurrent.Task;
+
+import java.lang.ref.WeakReference;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class ActorInfoContributorLifetimeExtension implements LifetimeExtension, InfoContributor
+{
+    private final Set<ActorInfoContributorReference> actorInfoContributorReferences = new HashSet<>();
+    private final OrbitRemoteReferenceService orbitRemoteReferenceService;
+    private final LifetimeExtension springLifetimeExtension;
+
+    ActorInfoContributorLifetimeExtension(final OrbitRemoteReferenceService orbitRemoteReferenceService,
+                                          final LifetimeExtension springLifetimeExtension)
+    {
+        this.orbitRemoteReferenceService = orbitRemoteReferenceService;
+        this.springLifetimeExtension = springLifetimeExtension;
+    }
+
+    @Override
+    public <T> T newInstance(final Class<T> concreteClass)
+    {
+        // In case Orbit chooses this extension as the "primary" LifetimeExtension, we need to redirect to the
```

Comment:
The problem is that the lifetime extension contains the newInstance interface, but it can't be passed through at all, so one is chosen randomly and you have no idea which of the lifetime extensions will actually be called. As such you have no way of knowing if SpringLifetimeExtension or ActorInfoContributorLifetimeExtension will be used.
Every other method in that interface does indeed act like a chain as you commented.
This has never been an issue before because we (and afaik no one else) has actually had 2 lifetime extensions before now.

The right solution is for newInstance to be in a different extension type, such as "ActorConstructorExtension" where the assumption is that you'd only have 1.